### PR TITLE
[OpenTelemetry] osquery-go trace instrumentation

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Pulls in changes from https://github.com/osquery/osquery-go/pull/110 and adds a small amount of instrumentation to trace codepaths from osquery-go to launcher and vice versa.

Screenshots below --

<details>
<summary>Call => Distributed => GetQueries</summary>

Screenshot 2023-06-30 at 11 20 04 AM

</details>

<details>
<summary>Call => Config</summary>

Screenshot 2023-06-30 at 11 21 18 AM

</details>

<details>
<summary>Call => Table => exec</summary>

Screenshot 2023-06-30 at 11 37 06 AM

</details>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module github.com/kolide/launcher
 
+replace github.com/osquery/osquery-go v0.0.0-20230627155640-6bcabfbc94ec => ../osquery-go
```
[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply go.mod:3]: Yes 🙂

File: cmd/grpc.ext/grpc.go [hunk 2]
```diff
@@ -151,11 +152,16 @@ type queryier struct {
 }
 
 func (q *queryier) Query(query string) ([]map[string]string, error) {
-	resp, err := q.client.Query(query)
+	ctx, span := traces.StartSpan(context.Background())
+	defer span.End()
```
[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply cmd/grpc.ext/grpc.go:156]: This was the one I was thinking of removing! I will update 👍

**Review by James-Pickett — APPROVED** [review]

File: pkg/traces/exporter/exporter.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/kolide/launcher/pkg/agent/types"
 	"github.com/kolide/launcher/pkg/backoff"
 	"github.com/kolide/launcher/pkg/osquery"
+	osquerygotraces "github.com/osquery/osquery-go/traces"
```
Comment (directionless) [inline pkg/traces/exporter/exporter.go:18]: Seeing this, makes me think we should put `SetTracerProvider` to being in the toplevel `osquery-go` library. and leave `traces` as effectively internal. But :shrug: maybe not important

Comment (RebeccaMahany) [reply pkg/traces/exporter/exporter.go:18]: I don't feel strongly about it either way

Comment (directionless) [reply pkg/traces/exporter/exporter.go:18]: I don't think it's worth fixing _now_, but it feels wrong. `traces` should really be an internal package, not part of the exposed API surface. But :shrug: we're most of the users, and I don't feel like we need to maintain some high standard of API consistency, so not worth it.

**Pull request merged**